mangaki/mangaki/wrappers/anilist.py

```diff
@@ -20,14 +21,155 @@ def to_python_datetime(date):
     >>> to_python_datetime('20150000')
     datetime.datetime(2015, 1, 1, 0, 0)
     """
-    date = date.strip()
+    date = str(date).strip()
     for fmt in ('%Y%m%d', '%Y%m00', '%Y0000'):
         try:
             return datetime.strptime(date, fmt)
         except ValueError:
             pass
     raise ValueError('no valid date format found for {}'.format(date))
 
+def to_anime_season(date):
+    """
+    Return the season corresponding to a date
+    >>> to_anime_season(datetime.datetime(2017, 3, 3, 0, 0))
+    'winter'
+    """
+    if 1 <= date.month <= 3:
+        return 'winter'
+    elif 4 <= date.month <= 6:
+        return 'spring'
+    elif 7 <= date.month <= 9:
+        return 'summer'
+    else:
+        return 'fall'
+
+
+class AniListWorks(Enum):
+    animes = 'anime'
+    mangas = 'manga'
+
+
+class AniListStatus(Enum):
+    aired = 'finished airing'
+    airing = 'currently airing'
+    published = 'finished publishing'
+    publishing = 'publishing'
+    anime_coming = 'not yet aired'
+    manga_coming = 'not yet published'
+    cancelled = 'cancelled'
+
+
+ANILIST_MEDIA_TYPES = ['TV',  'TV Short', 'Movie', 'Special', 'OVA', 'ONA', 'Music',
+                       'Manga', 'Novel', 'One Shot', 'Doujin', 'Manhua', 'Manhwa']
+
+
+class AniListMediaType(Enum):
+    tv = 0
+    tv_short = 1
+    movie = 2
+    special = 3
+    ova = 4
+    ona = 5
+    music = 6
+    manga = 7
+    novel = 8
+    one_shot = 9
+    doujin = 10
+    manhua = 11
+    manhwa = 12
+
+    def __str__(self):
+        return LIST_TYPES[self.value]
+
+
+class AniListEntry:
+    def __init__(self, anime_info, work_type: AniListWorks):
+        self.anime_info = anime_info
+        self.work_type = work_type
+        assert self.anime_info['series_type'] == work_type.value
```

Comment:
Why not raise an Exception rather than an assert?
`raise ValueError('AniList Data does not match `work_type` provided')` something like this.